aten/src/ATen/core/builtin_function.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include <ATen/core/dispatch/Dispatcher.h>
+#include <ATen/core/function.h>
+
+namespace torch {
+namespace jit {
+
+struct BuiltinOpFunction : public Function {
+  BuiltinOpFunction(
+      c10::QualifiedName qualname,
+      c10::FunctionSchema schema,
+      std::function<void(Stack&)> callable)
+      : name_(std::move(qualname)),
+        callable_(std::move(callable)),
+        schema_(std::move(schema)) {}
+
+  bool isGraphFunction() const override {
+    return false;
+  }
+
+  void run(Stack& stack) override {
+    callable_(stack);
+  }
+
+  void run(Stack&& stack) override {
+    callable_(stack);
+  }
+
+  at::IValue operator()(std::vector<at::IValue> stack, const Kwargs& kwargs)
+      override {
+    getSchema().checkAndNormalizeInputs(stack, kwargs);
+    callable_(stack);
+    return stack.front();
+  }
+
+  const c10::QualifiedName& qualname() const override {
+    return name_;
+  }
+
+  const std::string& name() const override {
+    return name_.name();
+  }
+
+  // if this isn't yet defined, run its method_creator function
+  void ensure_defined() override {
+    // nop
+  }
+
+  std::shared_ptr<Graph> graph() const override {
+    TORCH_INTERNAL_ASSERT(false);
+  }
+
+  std::shared_ptr<Graph> optimized_graph() const override {
+    TORCH_INTERNAL_ASSERT(false);
+  }
+
+  GraphExecutor& get_executor() override {
+    TORCH_INTERNAL_ASSERT(false);
+  }
+
+  const c10::FunctionSchema& getSchema() const override {
+    if (!adapted_schema_) {
```

Comment:
technically it's not threadsafe and also you'd need double-locking. But why not be safe and always take the mutex if it's not in the hot path?

Reply:
Technically that wouldn't be thread-safe either because we're returning a const ref here